web/experiments/feature_view.py

```diff
@@ -0,0 +1,240 @@
+            cv=LeaveOneOut(), param_grid={'alpha': np.linspace(start=0.0001, stop=16.0, num=nb_hyper)}, n_jobs=n_jobs,
+            scoring=make_scorer(f1_score))
+
+        gs.fit(X, y)
+        f1s = f1_score(y_true=y, y_pred=gs.predict(X))
+        cv_parms = gs.best_params_
+
+        logging.info('F1 score {} for feature {} with params {}'.format(f1s, f, cv_parms))
+
+        return f, f1s, cv_parms['alpha']
+
+    data = map(train, tqdm(features, desc='Train logreg for each feature'))
+
+    data = np.asarray(list(data))
+
+    features_f1_scored = dict(zip(data[:, 0], data[:, 1]))
+    features_cv_params = dict(zip(data[:, 0], data[:, 2]))
+
+    return features_f1_scored, features_cv_params
+
+
+def generate_figure(fs_id, fig_title, norms_path='./CSLB',
+                    fig_path='./cslb_feature_view_{:%d-%m-%Y_%H:%M}.png'.format(datetime.datetime.now()),
+                    show_visual=False):
+    """
+    This function generate picture for feature categories according to CLSB classification
+    :param fs_id: map between features and F1-score for logreg trained for it
+    :param norms_path: path to cslb dat files
+    :param fig_path: path where figure is saved. If is None figure is not stored
+    :param fig_title: added to title after CSLB
+    :param show_visual: show swarmplot using matplotlib
+    """
+
+    # for feature fit
+    categ = pd.read_csv(norms_path, sep='\t')
+    categ = categ[['feature type', 'concept', 'feature']]
+
+    def get_feature_category(word_str):
+        """Get all feature category"""
+        word_str = word_str.replace('_', ' ')
+        return categ.loc[categ['feature'] == word_str]['feature type'].unique().astype(str)
+
+    def binsort_features(feature_f1):
+        """Partition by feature type eg. 'encycopedic'"""
+        cat_val = defaultdict(list)
+        for f in feature_f1.keys():
+            for c in get_feature_category(f):
+                cat_val[c].append(feature_f1[f])
+
+        return cat_val
+
+    bf = binsort_features(fs_id)
+
+    df = pd.DataFrame.from_dict(bf, orient='index')
+
+    logging.info('Current matplotlib backend is {}'.format(plt.get_backend()))
+
+    # workaround for running matplotlib image generation without Xserver
+    if not show_visual:
+        logging.info('Switching matplotlib backend to Agg')
+        plt.switch_backend('agg')
+
+    sns.set_style("whitegrid")
+
+    tdf = df.transpose()
+    tdf.sort_index(axis=1, inplace=True)
+
+    sns.boxplot(data=tdf, boxprops=dict(alpha=0.2))
+    ax = sns.swarmplot(data=tdf)
+    plt.title('CSLB {}'.format(fig_title))
+
+    ax.set(xlabel='Feature category', ylabel='Feature fit score')
+
+    if fig_path is not None:
+        ax.get_figure().savefig(fig_path)
+        logging.info('Figure saved: {}'.format(fig_path))
+
+    if show_visual:
+        plt.show()
+
+    plt.close()
+
+
+def store_data(fs_id, fp_id, file_path):
```

Comment:
Make private. Remove test

Reply:
In current settings must be public.

Reply:
It is executed on data without reading .csv file after evaluate_cslb().

Reply:
Let's remove store_data function, let's move the part up to "df.sort_index(inplace=True)" back to evaluate_cslb, and let evaluate_cslb return this df *as the first item in the returned list* (along whatever it returns apart from it).

(Design motivation is again "less is more". This reduces cognitive load of the user. He runs evaluate_cslb, he gets back ready to use result: pd.dataframe, without need to do any merging, sorting, etc)

Reply:
Imho current state is sufficient - not like few months ago.